pkg/apis/awsprovider/v1alpha1/types_test.go

```diff
@@ -0,0 +1,162 @@
+				ToPort:                 80,
+				CidrBlocks:             []string{"10.20.30.0/24", "10.30.40.0/24", "172.16.12.0/24"},
+				SourceSecurityGroupIDs: []string{"sg-1234567", "sg-1236745"},
+			},
+			expected: false,
+		},
+		{
+			name: "Equal should return false for different FromPort",
+			inputIngressRule: &providerv1.IngressRule{
+				Description:            "allow access from vpc",
+				Protocol:               providerv1.SecurityGroupProtocolTCP,
+				FromPort:               80,
+				ToPort:                 80,
+				CidrBlocks:             []string{"10.20.30.0/24", "10.30.40.0/24", "172.16.12.0/24"},
+				SourceSecurityGroupIDs: []string{"sg-1234567", "sg-1236745"},
+			},
+			inputIngressRuleParam: &providerv1.IngressRule{
+				Description:            "allow access from vpc",
+				Protocol:               providerv1.SecurityGroupProtocolTCP,
+				FromPort:               443,
+				ToPort:                 80,
+				CidrBlocks:             []string{"10.20.30.0/24", "10.30.40.0/24", "172.16.12.0/24"},
+				SourceSecurityGroupIDs: []string{"sg-1234567", "sg-1236745"},
+			},
+			expected: false,
+		},
+		{
+			name: "Equal should return false for different ToPort",
+			inputIngressRule: &providerv1.IngressRule{
+				Description:            "allow access from vpc",
+				Protocol:               providerv1.SecurityGroupProtocolTCP,
+				FromPort:               0,
+				ToPort:                 80,
+				CidrBlocks:             []string{"10.20.30.0/24", "10.30.40.0/24", "172.16.12.0/24"},
+				SourceSecurityGroupIDs: []string{"sg-1234567", "sg-1236745"},
+			},
+			inputIngressRuleParam: &providerv1.IngressRule{
+				Description:            "allow access from vpc",
+				Protocol:               providerv1.SecurityGroupProtocolTCP,
+				FromPort:               0,
+				ToPort:                 443,
+				CidrBlocks:             []string{"10.20.30.0/24", "10.30.40.0/24", "172.16.12.0/24"},
+				SourceSecurityGroupIDs: []string{"sg-1234567", "sg-1236745"},
+			},
+			expected: false,
+		},
+		{
+			name: "Equal should return true for same CidrBlocks",
+			inputIngressRule: &providerv1.IngressRule{
+				Description:            "allow access from vpc",
+				Protocol:               providerv1.SecurityGroupProtocolTCP,
+				FromPort:               0,
+				ToPort:                 80,
+				CidrBlocks:             []string{"10.20.30.0/24", "10.30.40.0/24", "172.16.12.0/24"},
+				SourceSecurityGroupIDs: []string{"sg-1234567", "sg-1236745"},
+			},
+			inputIngressRuleParam: &providerv1.IngressRule{
+				Description:            "allow access from vpc",
+				Protocol:               providerv1.SecurityGroupProtocolTCP,
+				FromPort:               0,
+				ToPort:                 80,
+				CidrBlocks:             []string{"172.16.12.0/24", "10.20.30.0/24", "10.30.40.0/24"},
+				SourceSecurityGroupIDs: []string{"sg-1234567", "sg-1236745"},
+			},
+			expected: true,
+		},
+		{
+			name: "Equal should return true for same SourceSecurityGroupIDs",
```

Comment:
How is this test different from the previous one?

Also, looks like you'll have 8 cases returning false and only one case returning true.
For return false case:
* Different `SourceSecurityGroupIDs ` and `CidrBlocks` lengths case aren't covered.
* Different `SourceSecurityGroupIDs ` and `CidrBlocks` of equal length aren't covered.
Can you please add those?

Reply:
Shoot. I'll add those cases as well.